### ISSUE TITLE
Loading rpy2.ipython extension in notebook fails!!! 

### ISSUE THREAD

**Issue by rajaldebnath** [rpy2/rpy2#791]
` jupyter --version`

```
jupyter core     : 4.7.1
jupyter-notebook : 6.3.0
qtconsole        : 5.0.3
ipython          : 7.22.0
ipykernel        : 5.5.3
jupyter client   : 6.1.12
/usr/local/lib/python3.9/site-packages/jupyter_server/transutils.py:13: FutureWarning: The alias `_()` will be deprecated. Use `_i18n()` instead.
  warnings.warn(warn_msg, FutureWarning)
jupyter lab      : 3.0.12
nbconvert        : 6.0.7
ipywidgets       : 7.6.3
nbformat         : 5.1.3
traitlets        : 5.0.5
```

`pip show rpy2`

```
Name: rpy2
Version: 3.4.3
Summary: Python interface to the R language (embedded R)
Home-page: https://rpy2.github.io
Author: Laurent Gautier
Author-email: lgautier@gmail.com
License: GPLv2+
Location: /usr/local/lib/python3.9/site-packages
Requires: jinja2, cffi, pytz, tzlocal
Required-by:
```

<img width="912" alt="Screenshot 2021-04-03 at 11 28 57 AM" src="https://user-images.githubusercontent.com/9495745/113470001-d756c200-946f-11eb-9ed9-e0cb73a10611.png">

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-d8396572b2e5> in <module>
      1 # install rpy2 using pip, and then load it into the jupyter environment
----> 2 get_ipython().run_line_magic('load_ext', 'rpy2.ipython')

/usr/local/lib/python3.9/site-packages/IPython/core/interactiveshell.py in run_line_magic(self, magic_name, line, _stack_depth)
   2342                 kwargs['local_ns'] = self.get_local_scope(stack_depth)
   2343             with self.builtin_trap:
-> 2344                 result = fn(*args, **kwargs)
   2345             return result
   2346 

/usr/local/lib/python3.9/site-packages/decorator.py in fun(*args, **kw)
    212     else:
    213         def fun(*args, **kw):
--> 214             return caller(func, *(extras + args), **kw)
    215     fun.__name__ = func.__name__
    216     fun.__signature__ = inspect.signature(func)

/usr/local/lib/python3.9/site-packages/IPython/core/magic.py in <lambda>(f, *a, **k)
    185     # but it's overkill for just that one bit of state.
    186     def magic_deco(arg):
--> 187         call = lambda f, *a, **k: f(*a, **k)
    188 
    189         if callable(arg):

/usr/local/lib/python3.9/site-packages/IPython/core/magics/extension.py in load_ext(self, module_str)
     31         if not module_str:
     32             raise UsageError('Missing module name.')
---> 33         res = self.shell.extension_manager.load_extension(module_str)
     34 
     35         if res == 'already loaded':

/usr/local/lib/python3.9/site-packages/IPython/core/extensions.py in load_extension(self, module_str)
     78             if module_str not in sys.modules:
     79                 with prepended_to_syspath(self.ipython_extension_dir):
---> 80                     mod = import_module(module_str)
     81                     if mod.__file__.startswith(self.ipython_extension_dir):
     82                         print(("Loading extensions from {dir} is deprecated. "

/usr/local/Cellar/python@3.9/3.9.2_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py in import_module(name, package)
    125                 break
    126             level += 1
--> 127     return _bootstrap._gcd_import(name[level:], package, level)
    128 
    129 

/usr/local/Cellar/python@3.9/3.9.2_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/_bootstrap.py in _gcd_import(name, package, level)

/usr/local/Cellar/python@3.9/3.9.2_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/_bootstrap.py in _find_and_load(name, import_)

/usr/local/Cellar/python@3.9/3.9.2_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/_bootstrap.py in _find_and_load_unlocked(name, import_)

/usr/local/Cellar/python@3.9/3.9.2_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/_bootstrap.py in _load_unlocked(spec)

/usr/local/Cellar/python@3.9/3.9.2_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/_bootstrap_external.py in exec_module(self, module)

/usr/local/Cellar/python@3.9/3.9.2_4/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/_bootstrap.py in _call_with_frames_removed(f, *args, **kwds)

/usr/local/lib/python3.9/site-packages/rpy2/ipython/__init__.py in <module>
----> 1 from . import rmagic
      2 
      3 load_ipython_extension = rmagic.load_ipython_extension

/usr/local/lib/python3.9/site-packages/rpy2/ipython/rmagic.py in <module>
    828 __doc__ = __doc__.format(
    829                 R_DOC=' '*8 + RMagics.R.__doc__,
--> 830                 RPUSH_DOC=' '*8 + RMagics.Rpush.__doc__,
    831                 RPULL_DOC=' '*8 + RMagics.Rpull.__doc__,
    832                 RGET_DOC=' '*8 + RMagics.Rget.__doc__

TypeError: can only concatenate str (not "NoneType") to str
```

But switching to R kernel works fine!! What is going wrong here.. This is not a Conda installation of Jupyter or python
Thanks

**Comment by lgautier** [comment]
What do you get when running`python -m rpy2.situation` ?

**Comment by rajaldebnath** [comment]
Here is the output.


```
rpy2 version:
3.4.3
Python version:
3.9.2 (default, Mar 26 2021, 23:27:12) 
[Clang 12.0.0 (clang-1200.0.32.29)]
Looking for R's HOME:
    Environment variable R_HOME: None
    Calling `R RHOME`: /Library/Frameworks/R.framework/Resources
    Environment variable R_LIBS_USER: None
R's additions to LD_LIBRARY_PATH:

R version:
    In the PATH: R version 4.0.3 (2020-10-10) -- "Bunny-Wunnies Freak Out"
    Loading R library from rpy2: OK
Additional directories to load R packages from:
None
C extension compilation:
  include:
  ['/Library/Frameworks/R.framework/Resources/include']
  libraries:
  ['pcre2-8', 'lzma', 'bz2', 'z', 'icucore', 'dl', 'm', 'iconv']
  library_dirs:
  ['/usr/local/lib', '/usr/local/lib']
  extra_compile_args:
  []
  extra_link_args:
  ['-F/Library/Frameworks/R.framework/..', '-framework', 'R']
```

**Comment by lgautier** [comment]
This looks quite odd, and I don't remember anything like things reported.
Can you try to reinstall rpy2? Outside of the package's code being modified I don't see what could be the cause for the issue at the moment.

**Comment by krassowski** [comment]
Could it be due to `PYTHONOPTIMIZE` as in https://stackoverflow.com/a/37551689/6646912 ?

**Comment by yuxinchenNU** [comment]
I'm running into this issue too and my rpy2 version is 3.4.5. Any update on this? Thanks.

**Comment by lgautier** [comment]
The issue was fixed in PR #824. The fix will be in included in release 3.5.0.
